### DISCUSSION TITLE
How to get the date & time of the requested reviews of a pull request with the Rest API?

### DISCUSSION THREAD

**Discussion by aymericnovade** [community/community#38464] [Apps, API and Webhooks] [answered]
I'm trying to track reviews requested on pull requests, and I'd like to know how long ago a review was requested on a pull request. On the pull request page, we can see when someone was requested, but I'm not able to find this information with the Rest API
![image](https://user-images.githubusercontent.com/41351887/200246738-4f4a2ac7-cf4e-482f-bcbe-a283020d66d9.png)

**Comment by LangLangBart** [comment, marked as answer]
### REST API

- Could **not** find the answer to your request with `REST API`
  - [GitHub Docs - Get all requested reviewers for a pull request](https://docs.github.com/en/rest/pulls/review-requests#get-all-requested-reviewers-for-a-pull-request) 
  - the person to be requested is listed, but there is no information about when this event occurred.
  - if you find a way, please post your solution, alternatively see `GraphQL API` below.

---

### GraphQL API - timelineItems
Example: 
- repo: [cli/cli](https://github.com/cli/cli)
  - PR number: [5652](https://github.com/cli/cli/pull/5652)

#### GraphQL QUERY
```graphql
{
  repository(owner: "cli", name: "cli") {
    pullRequest(number: 5652) {
      timelineItems(itemTypes: REVIEW_REQUESTED_EVENT, first: 10) {
        nodes {
          ... on ReviewRequestedEvent {
            createdAt
            actor {
              login
            }
            requestedReviewer {
              ... on User {
                login
              }
            }
          }
        }
      }
    }
  }
}
```

#### RESULT
```json
{
  "data": {
    "repository": {
      "pullRequest": {
        "timelineItems": {
          "nodes": [
            {
              "createdAt": "2022-05-17T12:45:37Z",
              "actor": {
                "login": "heaths"
              },
              "requestedReviewer": null
            },
            {
              "createdAt": "2022-05-17T12:45:37Z",
              "actor": {
                "login": "heaths"
              },
              "requestedReviewer": {
                "login": "mislav"
              }
            }
          ]
        }
      }
    }
  }
}
```

**Reply by ghostinhershell** [reply]
Hey @aymericnovade,

If @LangLangBart's response here answered your question, can you mark their response as the answer to your question? Thanks!

**Comment by arvindcbh** [comment]
For anyone reading this later, I managed to get this information via the REST API.

Instead of the `requested_reviewers` endpoint, you'll instead have to use the `timeline` endpoint [documented here](https://docs.github.com/en/rest/issues/timeline?apiVersion=2022-11-28#list-timeline-events-for-an-issue). Filter for entries in `data` that have the `event` key set to `review_requested`

**Reply by chaitanyakapre051** [reply]
But it is only supported for issues, what if I need Pull request related events
